sonic_platform_base/sonic_xcvr/api/public/cmisTargetFWUpgrade.py

```diff
@@ -44,7 +43,38 @@
 
 class CmisTargetFWUpgradeAPI(CmisApi):
     def set_firmware_download_target_end(self, target):
-        return self.xcvr_eeprom.write(consts.TARGET_MODE, target)
+        """
+        Sets the target mode to the specified target.
+        If the target mode is set to a remote target, then the page select byte is set to 0.
+        Also, the remote target is then checked to ensure that its accessible.
+        In case of any error, the target mode is restored to E0.
+        Returns:
+            True if the target mode is set successfully, False otherwise.
+        """
+        try:
+            if not self.xcvr_eeprom.write(consts.TARGET_MODE, target):
+                return self._handle_error_and_restore_target_to_E0(
+                            "Failed to set target mode to {}".format(target))
+            if target != TARGET_E0_VALUE:
+                if not self.xcvr_eeprom.write(consts.PAGE_SELECT_BYTE, 0):
+                    return self._handle_error_and_restore_target_to_E0(
+                                "Failed to set page select byte to {}".format(target))
+                if not self._is_remote_target_accessible():
+                    return self._handle_error_and_restore_target_to_E0(
+                            "Remote target {} not accessible.".format(target))
+        except Exception as e:
+            return self._handle_error_and_restore_target_to_E0(
+                    "Exception occurred while setting target mode to {}: {}".format(target, repr(e)))
+
+        return True
+
+    def get_firmware_download_target_end(self):
```

Comment:
@mihirpat1 nit, rename to get_current_target_end()

Reply:
@mihirpat1 please check if this is tested for
1. remote target is already powered off
2. remote target is powered off after setting the target > 0

Reply:
@prgeor I have addressed this now. I haven't tested the power off scenario though.